src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -52,7 +63,7 @@ class Crystal::Doc::Generator
       Markdown.parse body, MarkdownDocRenderer.new(program_type, io)
     end
 
-    File.write "#{@dir}/index.html", MainTemplate.new(body, types, repository_name)
+    File.write "#{@dir}/index.html", MainTemplate.new(generate_flags(body), types, repository_name)
```

Comment:
I actually thing that instead of duplicating changes in this method, this can be replaced with a call to `doc(program_type, body)`.